code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/MessagingUtils.java

```diff
@@ -117,12 +118,27 @@ static boolean isEdgePersonalizationDecisionEvent(final Event event) {
                 MessagingConstants.EventSource.PERSONALIZATION_DECISIONS.equalsIgnoreCase(event.getSource());
     }
 
+    // ========================================================================================
+    // Scope retrieval
+    // ========================================================================================
+
     /**
-     * @param map A {@link Map} of any type.
-     * @return {@code boolean} indicating if the passed in {@code Map} is empty.
+     * @param propositionPayloads A {@link List<PropositionPayload>} containing propositions retrieved from AJO
+     * @return {@code String} containing the scope retrieved from the proposition payload details
      */
-    static boolean isMapNullOrEmpty(final Map map) {
-        return map == null || map.isEmpty();
+    static String getScope(final List<PropositionPayload> propositionPayloads) {
+        PropositionPayload propositionPayload = null;
+        if (propositionPayloads.size() > 0) {
+            propositionPayload = propositionPayloads.get(0);
+        }
+        return propositionPayload == null ? null : propositionPayload.propositionInfo.scope;
+    }
```

Comment:
i think this isn't used any more since we now do this check on a per-proposition basis when loading propositions to the rules engine